libraries/ag-charts-test/src/canvas/mock-canvas.ts

```diff
@@ -189,13 +139,13 @@ export function setup(opts: { width?: number; height?: number; document?: Docume
         if (element === 'canvas') {
             const mockedElement = realCreateElement.call(document, element, options) as HTMLCanvasElement;
 
-            const nextCanvas = mockCtx.canvasStack.shift() ?? createCanvas(width, height);
+            const nextCanvas = mockCtx.canvasStack.shift() ?? new ConfiguredCanvas(width, height);
             mockCtx.registerCanvasInstance(nextCanvas);
 
             proxyGetContext2D(mockCtx, nextCanvas, mockedElement);
 
-            mockedElement.toDataURL = (mimeType?: 'image/png') => {
-                return nextCanvas.toDataURL(mimeType ?? 'image/png');
+            mockedElement.toDataURL = (mimeType?: 'png') => {
+                return nextCanvas.toDataURLSync(mimeType ?? 'png');
```

Comment:
Their type is incorrect here

```suggestion
            mockedElement.toDataURL = (mimeType: 'image/png') => {
                return nextCanvas.toDataURLSync(mimeType.split('/')[1]);
```

Reply:
Good spot, applied